### ISSUE TITLE
[postgresql] The private DNS zone name  is not permitted to end with `azure.com` from August 28th, 2022

### ISSUE THREAD

**Issue by sinbai** [Azure/azure-rest-api-specs#20866]
When creating resource [postgresql flexible server ](https://github.com/Azure/azure-rest-api-specs/blob/7816ad3d42e9ff6717fed652c72f7fa609259f86/specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2021-06-01/postgresql.json#L42), the latest version of API requires that the name of [privateDnsZoneArmResourceId ](https://github.com/Azure/azure-rest-api-specs/blob/7816ad3d42e9ff6717fed652c72f7fa609259f86/specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2021-06-01/postgresql.json#L1762) has to end with `.postgres.database.azure.com`. However, per the [doc](https://learn.microsoft.com/en-us/azure/dns/private-dns-privatednszone#restrictions),  the private DNS zone name is not permitted to end with `azure.com` from August 28th, 2022. May I ask to update the validation to allow setting unreserved private DNS zone name for postgresql flexible server?

**Comment by navba-MSFT** [comment]
@sinbai 
Apologies for the late reply. Thanks for reaching out to us and reporting this issue. We are looking into this issue and we will provide an update.

**Comment by navba-MSFT** [comment]
@sinbai Apologies for the late reply. AFAIK, Currently we force the customers to have `postgres.database.azure.com` suffix for Prod regions. 
May I know if you are facing any issues while creating resource [postgresql flexible server ](https://github.com/Azure/azure-rest-api-specs/blob/7816ad3d42e9ff6717fed652c72f7fa609259f86/specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2021-06-01/postgresql.json#L42), using latest resource ? If Yes, please share the error details. Awaiting your reply.

**Comment by sinbai** [comment]
@navba-MSFT  thanks for your reply. Yes, we use the latest stable version 2021-06-01 API to create resource [postgresql flexible server](https://github.com/Azure/azure-rest-api-specs/blob/7816ad3d42e9ff6717fed652c72f7fa609259f86/specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2021-06-01/postgresql.json#L42), the error is as follows:

`  Code="PrivateDnsZoneHasWrongZoneSuffix" Message="The Private DNS Zone name provided is not valid. It has to end with .postgres.database.azure.com.`

I was not able to create the private DNS Zone name end with ".postgres.database.azure.com." as the private DNS zone name is not permitted to end with azure.com from August 28th, 2022. When creating a private DNS zone name end with ".postgres.database.azure.com", the error is as follows:

`Message="The Private DNS zone name 'acc220922020404116025.postgres.database.azure.com' is not permitted as it is a reserved Azure infrastucture domain and creating such zones may have unintended consequences for DNS resolution for services deployed in your virtual networks. Refer https://docs.microsoft.com/en-us/azure/dns/private-dns-privatednszone#restrictions for more details.`

**Comment by navba-MSFT** [comment]
@sinbai I had reached out to the Product Owners as this requires an indepth troubleshooting. Could you please let me know if you can create a a support ticket with these information ?. Then our support professionals can assist you further on this.
Please let me know your thoughts on this.

**Comment by sinbai** [comment]
@navba-MSFT thanks for your time. Could you explain where to create the "support ticket"?

**Comment by navba-MSFT** [comment]
@sinbai Open Azure Portal --> Navigate to your PostgreSql resource--> Select New Support Request as shown below:

![image](https://user-images.githubusercontent.com/57353862/195260470-2123b8c3-7649-4110-9d67-f506d1b76d47.png)


**Note:** If you don't have the right RBAC permissions to create a support ticket, then you may have to check with the subscription owner to create a support ticket on your behalf.

**Comment by sinbai** [comment]
@navba-MSFT thank you so much for your assitance.

**Comment by navba-MSFT** [comment]
@sinbai Please share the ticket # here once you have created it. So that I can check its progress.

**Comment by navba-MSFT** [comment]
@sinbai Please let us know if you were able to create a support ticket to troubleshoot this issue further.

**Comment by sinbai** [comment]
@navba-MSFT I didn't create a support ticket for this issue because the following error disappeared after I checked again. I don't know what happened to the private dsn zones backend. Anyway, I am unblocked by this issue. I will recreated a new issue if this happen again. I will close this issue and thanks again for your help.

`Message="The Private DNS zone name 'acc220922020404116025.postgres.database.azure.com' is not permitted as it is a reserved Azure infrastucture domain and creating such zones may have unintended consequences for DNS resolution for services deployed in your virtual networks. Refer https://docs.microsoft.com/en-us/azure/dns/private-dns-privatednszone#restrictions for more details.`

**Comment by justinmchase** [comment]
Why do you care what the zone is named? Its private it shouldn't matter at all.

**Comment by chrisegeba** [comment]
It would be great if you could share how you resolved it. I’m encountering conflicting information when trying to use a private DNS with a PostgreSQL Flexible Server. The documentation doesn’t seem to help, and responding to the error messages hasn’t worked either. Interestingly, when I configure it through the portal, it works with a DNS zone name that doesn’t align with the documentation.
